src/components/PermissionOption/index.tsx

```diff
@@ -8,8 +8,30 @@ export interface PermissionItem {
   description: string;
   permission: Permission;
   children?: PermissionItem[];
+  requires?: PermissionRequirement[];
 }
 
+interface PermissionRequirement {
+  permissions: Permission[];
+  type?: 'and' | 'or';
+}
+
+const hasRequirement = (
+  requirement: PermissionRequirement,
+  currentPermission: number
+): boolean => {
+  switch (requirement.type) {
+    case 'or':
+      return requirement.permissions.some(
+        (permission) => !!(permission & currentPermission)
+      );
+    default:
+      return requirement.permissions.every(
+        (permission) => !!(permission & currentPermission)
+      );
+  }
+};
```

Comment:
Why are we not just using `hasPermission` to check this?

Reply:
Fixed!